### PR TITLE
Fix invalid layer in value relation field

### LINKED ISSUE #3535: Crash at QgsVectorLayer::displayExpression()
We have a crash according to Google Play with the following backtrace:

```
*** *** *** *** *** *** *** *** *** *** *** *** *** *** *** ***
pid: 0, tid: 3217 >>> uk.co.lutraconsulting <<<

backtrace:
  #00  pc 0x0000000002a134ac  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libInput_arm64-v8a.so (QgsVectorLayer::displayExpression() const) (BuildId: 65afad8e1bbe51e700f8dc3497eb0af1d73204fc)
  #01  pc 0x0000000001c66420  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libInput_arm64-v8a.so (FeaturesModel::featureTitle(FeatureLayerPair const&) const+152) (BuildId: 65afad8e1bbe51e700f8dc3497eb0af1d73204fc)
  #02  pc 0x0000000001c658bc  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libInput_arm64-v8a.so (FeaturesModel::data(QModelIndex const&, int) const+176) (BuildId: 65afad8e1bbe51e700f8dc3497eb0af1d73204fc)
  #03  pc 0x0000000001cb9490  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libInput_arm64-v8a.so (RelationFeaturesModel::data(QModelIndex const&, int) const+276) (BuildId: 65afad8e1bbe51e700f8dc3497eb0af1d73204fc)
  #04  pc 0x00000000000a6a10  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6QmlModels_arm64-v8a.so (BuildId: 101cf89f1b254d264b32b55d61369fc76a2d7355)
  #05  pc 0x00000000000a6834  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6QmlModels_arm64-v8a.so (BuildId: 101cf89f1b254d264b32b55d61369fc76a2d7355)
  #06  pc 0x000000000030883c  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #07  pc 0x0000000000308778  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #08  pc 0x0000000000301f80  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (QV4::QObjectWrapper::getProperty(QV4::ExecutionEngine*, QV4::Heap::Object*, QObject*, QQmlPropertyData const*, QFlags<QV4::QObjectWrapper::Flag>)+940) (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #09  pc 0x00000000002e9a2c  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (QV4::Lookup::getterQObject(QV4::Lookup*, QV4::ExecutionEngine*, QV4::Value const&)+232) (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #10  pc 0x0000000000305120  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (QV4::QObjectWrapper::virtualResolveLookupGetter(QV4::Object const*, QV4::ExecutionEngine*, QV4::Lookup*)+728) (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #11  pc 0x00000000002e9994  /data/app/~~NB01d8lrGo1GxWxpKKYIgw==/uk.co.lutraconsulting-kTP8f55TuTOv3qu7aO1Jtg==/lib/arm64/libQt6Qml_arm64-v8a.so (QV4::Lookup::getterQObject(QV4::Lookup*, QV4::ExecutionEngine*, QV4::Value const&)+80) (BuildId: c82698348b8648048f614a6bf37b862623fb2b65)
  #12  pc 0x000000000000114c 
```

### PR DESCRIPTION
**Pull request by tomasMizera**
It seems there is some edge-case scenario where a referenced layer in the value relation field is not available (probably a missing layer or some other project misconfiguration ... I could not reproduce it with my projects)

Logs are indicating that `mTitleField` is empty, thus we continue to `FeaturesModel::FeatureTitle` with `pair->layer()` as nullptr. See https://github.com/MerginMaps/mobile/blob/dd05acc2fbc4a55e65715b799810e37368cf83b0/app/valuerelationfeaturesmodel.cpp#L87-L96

See bug report logs for more details -> Fixes #3535

### CODE REVIEW

**Review by VitorVieiraZ — APPROVED** [review]
Looking good to me. There is just an issue in the code layout check.

File: app/featuresmodel.cpp [hunk 1]
```diff
@@ -158,6 +158,12 @@ int FeaturesModel::rowCount( const QModelIndex &parent ) const
 
 QVariant FeaturesModel::featureTitle( const FeatureLayerPair &featurePair ) const
 {
+  if ( !featurePair.layer() || !featurePair.layer()->isValid() ) 
+  {
+    CoreUtils::log( QStringLiteral( "Features Model" ), QStringLiteral( "Received invalid feature layer pair!" ) );
+    return tr( "Unknown title" );
```
Comment (uclaros) [inline app/featuresmodel.cpp:164]: When `title` is empty we return `pair.feature().id()`
Should we not be returning the same for this case too?

Comment (tomasMizera) [reply app/featuresmodel.cpp:164]: Not sure, if the layer is invalid, I'd expect everything is broken 💥

Comment (uclaros) [reply app/featuresmodel.cpp:164]: OK, let's leave it like this. It might help identifying the edge case when this happens.

**Review by uclaros — APPROVED** [review]

**Pull request merged**